From: Yury Norov <yury.norov@gmail.com>
Subject: lib/group_cpus: fix initialization section in group_cpus_evenly()
Date: Wed, 10 Jan 2024 09:46:39 -0800

_free() parameters have to be initialized at declaration to avoid passing
uninitialized pointer to the destructor (unless such a parameter is
initialized before any possibility of return).

nr_present and nr_others are assigned by __group_cpus_evenly(), which may
return negative value in case of error.  So make it signed.

Link: https://lkml.kernel.org/r/ZZ7e3xbIKfVa3yqd@yury-ThinkPad
Signed-off-by: Yury Norov <yury.norov@gmail.com>
Reported-by: kernel test robot <lkp@intel.com>
Closes: https://lore.kernel.org/oe-kbuild-all/202312310941.FXBRzn8C-lkp@intel.com/
Reported-by: kernel test robot <lkp@intel.com>
Reported-by: Dan Carpenter <dan.carpenter@linaro.org>
Closes: https://lore.kernel.org/r/202401050452.IcgrcQON-lkp@intel.com/
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 lib/group_cpus.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/lib/group_cpus.c~lib-group_cpus-rework-group_cpus_evenly-fix
+++ a/lib/group_cpus.c
@@ -349,9 +349,9 @@ struct cpumask *group_cpus_evenly(unsign
 {
 	cpumask_var_t *node_to_cpumask __free(free_node_to_cpumask) = alloc_node_to_cpumask();
 	struct cpumask *masks __free(kfree) = kcalloc(numgrps, sizeof(*masks), GFP_KERNEL);
-	cpumask_var_t npresmsk __free(free_cpumask_var);
+	cpumask_var_t npresmsk __free(free_cpumask_var) = NULL;
 	cpumask_var_t nmsk __free(free_cpumask_var);
-	unsigned int curgrp, nr_present, nr_others;
+	int curgrp, nr_present, nr_others;
 
 	if (!masks || !node_to_cpumask || !alloc_cpumask_var(&nmsk, GFP_KERNEL)
 			|| !alloc_cpumask_var(&npresmsk, GFP_KERNEL))
_
